REGRESSION (287702@main): <input type=search> in vertical writing mode looks broken

lilyspiniolas opened the pull request:
#### 3dac33417503a1e3bf7a440e5cbe1afddab01524
<pre>
REGRESSION (287702@main): &lt;input type=search&gt; in vertical writing mode looks broken
<a href="https://bugs.webkit.org/show_bug.cgi?id=285553">https://bugs.webkit.org/show_bug.cgi?id=285553</a>
<a href="https://rdar.apple.com/142499953">rdar://142499953</a>

Reviewed by Abrar Rahman Protyasha and Aditya Keerthi.

Elements in vertical writing mode with appearance equal
to StyleAppearance::SearchField now once again get their
appearance updated to StyleAppearance::TextField regardless of
whether or not devolvable widgets are enabled or supported.

The new layout test checks to make sure the input is displayed
vertically by performing a mismatch check between a search input
partially obscured by a square, and the square only. When
displayed correctly, the tail of the input should protrude from
the square, causing a mismatch. When displayed incorrectly (i.e
the element does not have appearance equal to
StyleAppearance::TextField), the input will be entirely obscured
by the square, causing the two cases to match and thus the test
to fail.

* LayoutTests/fast/forms/vertical-writing-mode/input-search-expected-mismatch.html: Added.
* LayoutTests/fast/forms/vertical-writing-mode/input-search.html: Added.
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):

Canonical link: <a href="https://commits.webkit.org/288609@main">https://commits.webkit.org/288609@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/411c91bf4591478d4d89e6c6f0dc69b4917ce8d7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83913 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3531 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38214 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/88992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34921 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/85998 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3622 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11498 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/88992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/23105 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86959 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/2690 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/76249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/88992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2620 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/30471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/33970 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/73608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/31226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90363 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11178 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8088 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/73720 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11402 "Failed to checkout and rebase branch from PR 38682") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/72078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/72938 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18035 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/15908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/2513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11130 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/16602 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/10978 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14454 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12750 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (commented):
We should update the title to be prefixed with `REGRESSION (287702@main): `.

Also suggested a possible way we could test this over Slack.

pxlcoder commented on LayoutTests/fast/css/input-search-vertical.html:
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<head>
```
```suggestion
<!DOCTYPE html>
<html>
<head>
```

pxlcoder commented on LayoutTests/fast/css/input-search-vertical-expected-mismatch.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<head>
```
```suggestion
<!DOCTYPE html>
<html>
<head>
```

pxlcoder reviewed (commented):
You can move the test to `LayoutTests/fast/forms/vertical-writing-mode` and drop `-vertical` from the test name.

I'd also briefly describe how the test works in the commit message, since it's not as straightforward as other tests.

lilyspiniolas committed:
REGRESSION (287702@main): <input type=search> in vertical writing mode looks broken
https://bugs.webkit.org/show_bug.cgi?id=285553
rdar://142499953

Reviewed by Abrar Rahman Protyasha and Aditya Keerthi.

Elements in vertical writing mode with appearance equal
to StyleAppearance::SearchField now once again get their
appearance updated to StyleAppearance::TextField regardless of
whether or not devolvable widgets are enabled or supported.

The new layout test checks to make sure the input is displayed
vertically by performing a mismatch check between a search input
partially obscured by a square, and the square only. When
displayed correctly, the tail of the input should protrude from
the square, causing a mismatch. When displayed incorrectly (i.e
the element does not have appearance equal to
StyleAppearance::TextField), the input will be entirely obscured
by the square, causing the two cases to match and thus the test
to fail.

* LayoutTests/fast/forms/vertical-writing-mode/input-search-expected-mismatch.html: Added.
* LayoutTests/fast/forms/vertical-writing-mode/input-search.html: Added.
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):

Canonical link: https://commits.webkit.org/288609@main
